lib/onebox/engine/tiktok_onebox.rb

```diff
@@ -0,0 +1,58 @@
+# frozen_string_literal: true
+
+module Onebox
+  module Engine
+    class TiktokOnebox
+      include Engine
+      include StandardEmbed
+
+      matches_regexp(%r{^https?://((?:m|www)\.)?tiktok\.com(?:/@(.+)\/video/|/v/)\d+(/\w+)?/?})
+      requires_iframe_origins "https://www.tiktok.com"
+      always_https
+
+      def placeholder_html
+        <<-HTML
+          <img
+            src="#{oembed_data.thumbnail_url}"
+            title="#{oembed_data.title}"
+            style="
+              max-width: #{oembed_data.thumbnail_width / 2}px;
+              max-height: #{oembed_data.thumbnail_height / 2}px;"
+          >
+        HTML
+      end
+
+      def to_html
+        height = oembed_data.thumbnail_width >= oembed_data.thumbnail_height ? 727 : 742
+
+        <<-HTML
+          <iframe
+            class="tiktok-onebox"
+            src="https://www.tiktok.com/embed/v2/#{oembed_data.embed_product_id}"
+            sandbox="allow-popups allow-popups-to-escape-sandbox allow-scripts allow-top-navigation allow-same-origin"
+            frameborder="0"
+            seamless="seamless"
+            scrolling="no"
+            style="
+              min-width: 332px;
+              height: #{height}px;
+              border-top: 3px solid #fff;
+              border-radius: 9px;
+              background-color: #fff;
```

Comment:
Can we move these styles to a stylesheet and use color variables? I suspect the white border here is not going to fit great with dark color schemes.

Reply:
I included this as inline because it's a bit of a hack to make their inframe work. The issue is that the embed has a white div you cannot modify.

![Screenshot from 2023-01-23 13-01-32](https://user-images.githubusercontent.com/66427541/214093655-a911da15-8621-4d32-a0d0-977a8238e15a.png)

So to make it look better, I had to shrink it down and add a border at the top to make everything even. 

![bitmapewed](https://user-images.githubusercontent.com/66427541/214093708-15e84b02-1122-4dd1-9610-cd4312833ca0.png)

This works great, but when the embed is loading, you can see the border for a brief second when the background is other than white. That's why the background of the iframe is set to white. Is not to match the post background, but to match their iframe color, which is always white

![bitmap](https://user-images.githubusercontent.com/66427541/214093749-19be4b17-d558-423c-b161-ff8656242a4d.png)